packages/cli/src/__tests__/external-hooks.test.ts

```diff
@@ -0,0 +1,111 @@
+import { mock } from 'jest-mock-extended';
+import type { ErrorReporter } from 'n8n-core';
+import type { IWorkflowBase } from 'n8n-workflow';
+import { ApplicationError } from 'n8n-workflow';
+
+import config from '@/config';
+import type { CredentialsRepository } from '@/databases/repositories/credentials.repository';
+import type { SettingsRepository } from '@/databases/repositories/settings.repository';
+import type { UserRepository } from '@/databases/repositories/user.repository';
+import type { WorkflowRepository } from '@/databases/repositories/workflow.repository';
+import { ExternalHooks } from '@/external-hooks';
+
+jest.mock('@/config', () => ({
+	getEnv: jest.fn(),
+}));
+
+describe('ExternalHooks', () => {
+	const errorReporter = mock<ErrorReporter>();
+	const userRepository = mock<UserRepository>();
+	const settingsRepository = mock<SettingsRepository>();
+	const credentialsRepository = mock<CredentialsRepository>();
+	const workflowRepository = mock<WorkflowRepository>();
+
+	const workflowData = mock<IWorkflowBase>({ id: '123', name: 'Test Workflow' });
+	const hookFn = jest.fn();
+
+	let externalHooks: ExternalHooks;
+
+	beforeEach(() => {
+		jest.resetAllMocks();
+		externalHooks = new ExternalHooks(
+			errorReporter,
+			userRepository,
+			settingsRepository,
+			credentialsRepository,
+			workflowRepository,
+		);
+	});
+
+	describe('init()', () => {
```

Comment:
The DB repositories are part of the public interface. Can we test that the repositories are present when we instantiate external hooks? And that the keys used to reference them are the current ones?